whipper/common/encode.py

```diff
@@ -89,3 +90,66 @@ def _tag(self):
         w.save()
 
         self.stop()
+
+
+class EmbedPictureTask(task.Task):
+    description = 'Embed picture to FLAC'
+
+    def __init__(self, track_path, cover_art_path):
+        self.track_path = track_path
+        self.cover_art_path = cover_art_path
+
+    def start(self, runner):
+        task.Task.start(self, runner)
+        self.schedule(0.0, self._embed_picture)
+
+    def _make_flac_picture(self, cover_art_filename):
+        """
+        Given a path to a jpg/png file, return a FLAC picture for embedding.
+
+        The embedding will be performed using the mutagen module.
+
+        :param cover_art_filename: path to cover art image file
+        :type cover_art_filename: str
+        :returns: a valid FLAC picture for embedding
+        :rtype: mutagen.flac.Picture or None
+        """
+        if not cover_art_filename:
+            return
+
+        from PIL import Image
+
+        im = Image.open(cover_art_filename)
+        if im.format == 'JPEG':
+            mime = 'image/jpeg'
+        elif im.format == 'PNG':
+            mime = 'image/png'
+        else:
+            # we only support png and jpeg
+            logger.warning("no cover art will be added because the fetched "
+                           "image format is unsupported")
+            return
+
+        pic = Picture()
+        with open(cover_art_filename, 'rb') as f:
+            pic.data = f.read()
+
+        pic.type = PictureType.COVER_FRONT
+        pic.mime = mime
+        pic.width, pic.height = im.size
+        if im.mode not in ('P', 'RGB', 'SRGB'):
+            return
```

Comment:
I think a logger statement should be added before the `return` statement. Something like this:

```python
            logger.warning("no cover art will be added because the fetched "
                           "image mode is unsupported")
```